index.html

```diff
@@ -12763,14 +12763,19 @@ <h2>Excluding Elements from the Accessibility Tree</h2>
 	  <h2>Including Elements in the Accessibility Tree</h2>
           <p>If not already excluded from the accessibility tree per the rules above in <a href="#tree_exclusion">Excluding Elements in the Accessibility Tree</a>, user agents MUST provide an <a>accessible object</a> in the <a>accessibility tree</a> for <abbr title="Document Object Model">DOM</abbr> <a>elements</a> that meet any of the following criteria:</p>
           <ul>
-            <li>Text nodes</li>
-            <li>Elements that may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> <a>event</a>.</li>
-            <li>Elements that are focusable, or have an ID <a>attribute</a> and an ancestor with the <pref>aria-activedescendant</pref> attribute that matches the implicit or explicit <a>semantics</a> of the required context role. In either case, the element may receive focus and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> focus event.</li>
+            <li>Elements that are not <a>hidden</a> and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> <a>event</a>:
+              <ul>
+                <li>Elements that are focusable may receive focus and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> focus event.</li>
+                <li>Elements that have an ID <a>attribute</a>, and are <a>owned</a> by, controlled by, or have an ancestor with the <pref>aria-activedescendant</pref> attribute, and that match the implicit or explicit <a>semantics</a> of the required context role, may receive focus and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> focus event.</li>
```

Comment:
This text doesn't quite capture the requirements. The element with the ID is not the element that is controlled; its container is controlled, and that is only in the case where the container is not focusable, i.e., a textbox or combobox is focusable and controls the container.
And, the role of the element with the ID must match a required context role of the container.
Additionally, the container must support aria-active descendant.

Per above, we should also remove the reference to the focus event.

I suggest the following.

```suggestion
                <li>Elements that have an ID <a>attribute</a>, and are contained in or <a>owned</a> by an element that supports the <pref>aria-activedescendant</pref> property, and have an implicit or explicit role that matches the required context role of that containing element, and that containing element is either focusable and has the <pref>aria-activedescendant</pref> property or is controled bby an element that is focusable and has the <pref>aria-activedescendant<pref> property.</li>
```

Reply:
@mcking65 
Wow, that sentence has a lot of words. :)
Would it be any clearer to use specific roles and split it into 2 separate points as follows?

- Elements that have an ID attribute, and are contained in or owned by a focusable composite, group or application that has the aria-activedescendant property, and have an implicit or explicit role that matches the required context role of their containing element.

- Elements that have an ID attribute, and are contained in or owned by a listbox, tree, grid or dialog that is controlled by a focusable textbox or combobox that has the aria-activedescendant property, and have an implicit or explicit role that matches the required context role of their containing element.

---

Alternatively, if a single, more generic, sentence is preferred, then perhaps:

- Elements that have an ID attribute, and are contained in or owned by an element that supports and has the aria-activedescendant property, and have an implicit or explicit role that matches the required context role of their containing element, and the containing element is focusable or is controlled by an element that is focusable.

Reply:
@mcking65 For now, I have committed the "single sentence" option (2nd alternative, above).
Please let me know if you prefer any of the other alternatives above.